edit social media links with custom url links and add the sharing to …

ebuka-nwabueze opened the pull request:
…featured event card

ebuka-nwabueze committed:
edit social media links with custom url links and add the sharing to featured event card

ebuka-nwabueze committed:
add format time util and training id

ebuka-nwabueze committed:
- increase modal height
- Reduce modal width
- add the required field for input fields

ebuka-nwabueze committed:
add the leave message on modal

ebuka-nwabueze committed:
add first part of the filter section

ebuka-nwabueze committed:
add the filter panel

ebuka-nwabueze committed:
add sample data

ebuka-nwabueze committed:
add css for filter

ebuka-nwabueze committed:
add the selected filter tiles

ebuka-nwabueze committed:
rename selected filter to filter tiles

ebuka-nwabueze committed:
comment the filter list

Ozoemenamsolomon commented on src/components/FeaturedEvents.tsx:
```diff
@@ -4,15 +4,22 @@ import styled from 'styled-components';
 import FeaturedEventCard from './FeaturedEventCard';
 import { Container } from './styled';
+import FeaturedEventsFilter from './FeaturedEventsFilter';
+interface FeaturedEventsProp {
```
When do you choose `type` over `interface` and vice-versa?

ebuka-nwabueze replied:
I have unified it to type in this case

Ozoemenamsolomon replied:
I didn't say it's wrong, I'm just trying to understand

ebuka-nwabueze replied:
Oh yeah! Normally i use interface when i want to do some sort of inheritance with the props but in this cases, there isn't any. As you pointed it out, i decided to revert it to type for simplicity.

Ozoemenamsolomon replied:
Okay, there's no right or wrong but consistency is important

ebuka-nwabueze committed:
add react children to the checkbox group

ebuka-nwabueze committed:
redesign the filter panel

ebuka-nwabueze committed:
center the filter

ebuka-nwabueze committed:
change the category list to use filterList

ebuka-nwabueze committed:
change the date state to use filterlist

ebuka-nwabueze committed:
remove unused variables

ebuka-nwabueze committed:
adding initial filter list in the url

Ozoemenamsolomon committed:
Merge branch 'main' into feature/update-training-dialog

ebuka-nwabueze committed:
replace dummydata with db data

ebuka-nwabueze committed:
change image on book now to dynamic url

ebuka-nwabueze committed:
add the function to return the unique values from the training data

ebuka-nwabueze committed:
make the options display of flex

ebuka-nwabueze committed:
update the query param update links

ebuka-nwabueze committed:
update the query params to use dynamic values and also update the filter list

ebuka-nwabueze committed:
add the filtering logic

Ozoemenamsolomon committed:
Merge branch 'main' into feature/update-training-dialog

ebuka-nwabueze committed:
fix bug on clearing filter and query params

ebuka-nwabueze committed:
remove filter button and related onClick functions

ebuka-nwabueze committed:
add date sorting to the training data

ebuka-nwabueze committed:
change the view more events to use the filtered data

ebuka-nwabueze committed:
change date filter to use earliest date

ebuka-nwabueze committed:
add the date filter functionality

ebuka-nwabueze committed:
fix the date start and end on page mount

ebuka-nwabueze committed:
fix some css styling

add message to whatsapp link

ebuka-nwabueze committed:
add the href to the call link in home page

ebuka-nwabueze committed:
add promo data fetching props

ebuka-nwabueze committed:
add promo code discount

ebuka-nwabueze committed:
adjust css styles

Ozoemenamsolomon commented on src/components/FeaturedEventsFilter.tsx:
```diff
@@ -0,0 +1,464 @@
+	}
+`;
+
+const DatePickerWrapper = styled.div`
```
This is the same way I expected you'd style the radix modal instead of creating a CSS module

ebuka-nwabueze replied:
I tried it but couldn't get it correctly. It has specific style inheritance. The CSS module was quite easier to target the inherited styles. That's why. i will still use the radix again in the future for a feature, i'll try again to figure it out, then i can re-implement this.

Ozoemenamsolomon commented on src/components/FeaturedEventDialog.tsx:
```diff
@@ -23,13 +23,9 @@ interface FormDataType {
-	training,
-}) => {
+const FeaturedEventDialog: React.FC<Props> = ({ training }) => {
 	const [numPeople, setNumPeople] = useState(1);
```
Coming back to this number of people, use [derived state](https://www.linkedin.com/pulse/why-you-should-use-derive-state-react-better-rohan-vania/)

ebuka-nwabueze replied:
I still don't understand how you mean here. 
At the start of the opening of the booking dialog, the number of people is one by default. The numOfPeople changes based on how many people/ participants are to book for a training.

Could you emphasize how the derive state can be added here. I use this to decide the new price of the training, people attending and so on.

Ozoemenamsolomon replied:
I'll refactor it later after the branch is merged

ebuka-nwabueze replied:
Ok. No problem. I'll keep an eye on it to see the refactored version.

Ozoemenamsolomon commented on src/components/CallButton.tsx:
```diff
@@ -1,22 +1,22 @@
 import styled from 'styled-components';
-import { CTAFlex } from './CTA';
```
If the CTAFlex is not been used again, make sure to get rid of unused code

ebuka-nwabueze replied:
I've removed all references.

Ozoemenamsolomon commented on src/components/FeaturedEventDialog.tsx:
Nice, I see you removing comments that might not be used  again 👍🏽

Ozoemenamsolomon commented on src/components/FeaturedEventsFilter.tsx:
```diff
@@ -0,0 +1,464 @@
+
+	const router = useRouter();
+
+	const handleQueryParamsOnChange = useCallback(
```
why did you choose to use `useCallback` here and below? Just asking.

ebuka-nwabueze replied:
For it to return a optimised function, rather than the normal function. This way it would not be calling it multiple times unnecessarily in the background.

Ozoemenamsolomon replied:
Okay, it's basically just because of the dependence on the router. To prevent unnecessary **recreation** of the functions when the router object hasn't changed. Well, I usually just create the function outside and pass the router object as a parameter to the function.

Ozoemenamsolomon reviewed (commented):
You've set the pull request to ready but I see you're still pushing changes, is it really ready?

ebuka-nwabueze committed:
create promo code discount

ebuka-nwabueze committed:
update code based on PR review

ebuka-nwabueze committed:
add date check for the promo

ebuka-nwabueze commented:
I'm done adding changes for now. If all looks good, you can merge.

Ozoemenamsolomon commented on src/components/FeaturedEventCard.tsx:
```diff
@@ -5,34 +5,79 @@ import Location from '@assets/new/icons/Location';
+
+	const redeemPromoCode = () => {
+    // Check if the entered promo code exists in the promo list
+    const promo = promoData.find((promo) => promo.promo_code === promoCode);
```
This promo check should be done in an API route.

ebuka-nwabueze replied:
I've updated it to use api now

ebuka-nwabueze committed:
change the promo to use api routes

ebuka-nwabueze committed:
update promoIsValid in promo and the state in event card

Ozoemenamsolomon committed:
DB handles all checks
- if DB return no data, then promocode is invalid
- this commit is just for clarification

Ozoemenamsolomon committed:
clean up after clarification
- prevent api calls for empty promoCode

Ozoemenamsolomon committed:
final clean up

Ozoemenamsolomon reviewed (approved):
I'm gonna approve the PR now, but I think the following are note-worthy:
- I strongly believe each promo should be associated with an event or maybe more
- Sending price to the API and promoCode isn't really practical, as anyone could send any price to the DB at the moment as it's an unprotected route. I'd suggest sending the event Identifier instead

Ozoemenamsolomon committed:
Merge branch 'main' into feature/update-training-dialog
